neuroinfer/code/run_bayesian.py

```diff
@@ -193,6 +204,70 @@ def get_atlas_coordinates_json(json_path):
     return coordinates_atlas
 
 
+def is_visited(current_coord, visited_coords):
+    """
+    Check if a coordinate has been visited.
+
+    Args:
+        current_coord (tuple): Tuple containing x, y, and z coordinates.
+        visited_coords (ndarray): 3D array of visited coordinates.
+
+    Returns:
+        bool: True if the coordinate has been visited, False otherwise.
+    """
+    x, y, z = current_coord
+    if (
+        0 <= x < visited_coords.shape[0]
+        and 0 <= y < visited_coords.shape[1]
+        and 0 <= z < visited_coords.shape[2]
+    ):
+        return visited_coords[x, y, z] == 1
+    else:
+        return False
+
+
+def update_visited_coord(visited_coords, x_norm, y_norm, z_norm, padding):
+    """
+    Update the array of visited coordinates with the given normalized coordinates.
+
+    Args:
+        visited_coords (ndarray): 3D array of visited coordinates.
+        x_norm (int): Normalized x-coordinate.
+        y_norm (int): Normalized y-coordinate.
+        z_norm (int): Normalized z-coordinate.
+        padding (int): Padding value to be added around the visited coordinates.
+
+    Returns:
+        None
+    """
+    x_min = max(0, x_norm - padding)
+    x_max = min(visited_coords.shape[0], x_norm + padding + 1)
+    y_min = max(0, y_norm - padding)
+    y_max = min(visited_coords.shape[1], y_norm + padding + 1)
+    z_min = max(0, z_norm - padding)
+    z_max = min(visited_coords.shape[2], z_norm + padding + 1)
+    visited_coords[x_min:x_max, y_min:y_max, z_min:z_max] = 1
```

Comment:
this will occupy a cubic space of 1s. in order to have it spheric we can recycle  np.linalg.norm